rules/aip0203/field_behavior_required.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package aip0203
+
+import (
+	"fmt"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var requiredOneOfFieldBehavior = stringset.New(
+	"OPTIONAL", "REQUIRED", "OUTPUT_ONLY",
+)
+
+var fieldBehaviorRequired = &lint.FieldRule{
+	Name: lint.NewRuleName(203, "field-behavior-required"),
```

Comment:
I think it is pretty obvious that this will result in many many more findings for folks, especially with the cascading affect of 1 field being added, field_behavior is recommended, then it is added, then the following finding from the  "consistency" rule e.g. "all fields should have field_behavior or none", followed by any other ApiLinter rule that fires on the fields being changed just to add field_behavior.

I'm not going to lie, I'm nervous about the impact that will have on users and on the perception of ApiLinter.

Reply:
I'm definitely open to suggestions. 

I thought about the cascading effect issue, and thought about maybe adding to
the error message that one of each must be set?

I think we agree that having these fields be required for new fields is better
than not having them. And hopefully, since findings only affect modified lines,
we should see gradual introduction to this new behavior.

The only issue I can think of is that the error is annoying enough that we don't
want to introduce it altogether. Which, I think based on the impact to clients,
is a price we have to pay.

I'm open to a high-bandwidth discussion on this too. WDYT?

Reply:
I think we should add this rule, and do one of the follow:
- refine the scope to only apply to Resource & Request messages (you've kind of started on this already) -> this reduces the blast radius
- remove the "optional-consistency" [rule](https://linter.aip.dev/203/optional-consistency) (I guess its just for OPTIONAL, which is better than the alternative) -> eliminates the cascading issue
- do nothing different, we really do want anything & everything annotated, see what happens

Reply:
I'll likely go for:

- refine scope to resource + request messages. I *think* I can do this by making it a method Lint and grabbing the request time. If we recursively dig through fields and look at their structs it should work.
- remove optional consistency. 

If that sounds good.

Reply:
Yeah I think that sounds good.

Reply:
done!